src/Controller/Component/SchemaComponent.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace App\Controller\Component;
+
+use Cake\Cache\Cache;
+use Cake\Controller\Component;
+use Cake\Utility\Hash;
+
+/**
+ * Handles JSON SCHEMA of objects and resources
+ *
+ */
+class SchemaComponent extends Component
+{
+    /**
+     * Cache config name for type schemas.
+     *
+     * @var string
+     */
+    const CACHE_CONFIG = '_schema_types_';
+
+    /**
+     * BEdita4 API client
+     *
+     * @var \App\Model\API\BEditaClient
+     */
+    protected $apiClient = null;
+
+    /**
+     * Type name
+     *
+     * @var string
+     */
+    protected $type = null;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected $_defaultConfig = [
+        'apiClient' => null,
+    ];
+
+    /**
+     * {@inheritDoc}
+     */
+    public function initialize(array $config)
+    {
+        $this->apiClient = Hash::get($config, 'apiClient', null);
+        $this->type = Hash::get($config, 'type', null);
```

Comment:
`null` is not necessary as third parameter here: it is assumed by default.

In general, everything in this method is not necessary, since these values could be obtained with `$this->getConfig('apiClient')` and `$this->getConfig('type')`, so there's no need to persist these values as instance attributes. Also, this might cause unexpected behaviors if the user invokes `$this->Schema->setConfig()` in the controller, because the values are not sync'ed.